Content.Server/Crayon/CrayonSystem.cs

```diff
@@ -125,6 +132,54 @@ private void OnCrayonBoundUIColor(EntityUid uid, CrayonComponent component, Cray
 
     }
 
+    private void OnCrayonBoundUIRotation(EntityUid uid, CrayonComponent component, CrayonRotationMessage args)
+    {
+        component.Rotation = args.Rotation;
+        Dirty(uid, component);
+    }
+
+    private void OnCrayonBoundUIPreviewMode(EntityUid uid, CrayonComponent component, CrayonPreviewModeMessage args)
+    {
+        if (TryComp<HandsComponent>(args.Actor, out var hands) &&
+            TryComp<CrayonComponent>(hands.ActiveHandEntity, out var crayon) &&
+            hands.ActiveHandEntity == uid)
+        {
+            // Only toggle the overlay if the user is holding a crayon in their active hand
+            // and check if it is the same crayon that sent the request
+            component.PreviewMode = args.PreviewMode;
+            Dirty(uid, component);
+            RaiseNetworkEvent(new CrayonOverlayUpdateEvent(component.SelectedState, component.Rotation, component.Color, component.PreviewMode));
+        }
+        else
+        {
+            // failed to enable, reset button toggle
+            _uiSystem.SetUiState(uid, SharedCrayonComponent.CrayonUiKey.Key, new CrayonBoundUserInterfaceState(component.SelectedState, component.SelectableColor, component.Color, component.Rotation, component.PreviewMode));
+        }
+    }
+
+    private void OnBuiClosed(EntityUid uid, CrayonComponent component, BoundUIClosedEvent args)
+    {
+        DisablePreviewMode(uid, component);
+    }
+
+    private void OnHandDeselected(EntityUid uid, CrayonComponent component, ref HandDeselectedEvent args)
```

Comment:
```suggestion
    private void OnHandDeselected(Entity<CrayonComponent> ent, ref HandDeselectedEvent args)
```
use the `Entity<Component>` pattern
same for all the other methods here
In case you don't know about that, I got a writeup here https://hackmd.io/@Slart/S1hsoGFm1l